### PR TITLE
Adding Right-to-Left support for template

### PR DESCRIPTION
**Pull request by hamidrezakp**
Adding Right-to-Left support for template. This is not a full solution to RTL problem because some English paragraphs are also Right-to-Left and must be fixed.

Now every post that want to be RTL must add an `rtl = true` in the `extra` section of post.

### CODE REVIEW

**Comment by phil-opp** [comment]
Thanks! Using an `extra.rtl` setting for specifying the text direction is a good idea.

Instead of adding a separate CSS file and applying the `rtl` layout to all text, I would prefer solving this via classes. We could add a `right-to-left` classes in our HTML and then use a rule like this in our existing `main.css`:

```css
.right-to-left p, .right-to-left h1, … {
	direction: rtl;
	font-family: "Vazir, Sahel, Yekan";
}
```

This should allow us to set the text direction per block. In the `page.html` template, we could then apply the text direction only to the `page.content` and `page.title`.

We could also add a `left-to-right` class that can be used for overriding the text direction within `right-to-left` blocks. Since CSS always applies the most specific rule, this should allow us to arbitrarily nest ltr and rtl blocks.

**Comment by hamidrezakp** [comment]
Yeah you are right, using a `right-to-left` class in `main.css` is way better. also the `left-to-right` class is so good because we have some `code` blocks inside `p` elements and we can nicely use nested `rtl` and `ltr` in text.

However i don't know how we can apply `right-to-left` classes to `page.content`. i searched Zola documents but didn't find anything useful. can you give and example?
thanks

**Comment by phil-opp** [comment]
We could just add an additional `<div>` element around it, to which we can add the class then.

**Comment by hamidrezakp** [comment]
Ok, i have done changes based on `Classes`. let me know if there is any other issue.

File: blog/templates/second-edition/page.html [hunk 1]
```diff
@@ -33,12 +33,20 @@ <h2>Table of Contents</h2>
...
     <h1>{{ page.title }}</h1>
     <time datetime="{{ page.date | date(format="%Y-%m-%d") }}" class="post-date">
         {{ page.date | date(format="%b %d, %Y") }}
         {% if page.extra.updated %} (updated on {{ page.extra.updated | date(format="%b %d, %Y") }}) {% endif %}
     </time>
 
+    {% if page.extra.rtl %}
+	</div>
+    {% endif %}
+
```
Comment (phil-opp) [inline blog/templates/second-edition/page.html:49]: We can simplify this by making only the class of the div conditional:

```suggestion
    <div class="{% if page.extra.rtl %}right-to-left{% endif %}">
    <h1>{{ page.title }}</h1>
    <time datetime="{{ page.date | date(format="%Y-%m-%d") }}" class="post-date">
        {{ page.date | date(format="%b %d, %Y") }}
        {% if page.extra.updated %} (updated on {{ page.extra.updated | date(format="%b %d, %Y") }}) {% endif %}
    </time>
    </div>
```

File: blog/templates/second-edition/page.html [hunk 2]
```diff
@@ -67,8 +75,16 @@ <h1>{{ page.title }}</h1>
...
+    {% if page.extra.rtl %}
+	<div class="right-to-left">
+    {% endif %}
+
     {{ page.content | replace(from="<!-- toc -->", to=macros::toc(toc=page.toc))  | safe }}
 
+    {% if page.extra.rtl %}
+	</div>
+    {% endif %}
+
```
Comment (phil-opp) [inline blog/templates/second-edition/page.html:87]: Similar to above:

```suggestion
    <div class="{% if page.extra.rtl %}right-to-left{% endif %}">
    {{ page.content | replace(from="<!-- toc -->", to=macros::toc(toc=page.toc))  | safe }}
    </div>
```

**Review by phil-opp — COMMENTED** [review]
Thanks for the update! I left some simplification suggestions inline:

**Comment by hamidrezakp** [comment]
Thanks a lot for suggestions!, now code is cleaner and so straight forward.

**Comment by phil-opp** [comment]
Thanks!

**Pull request merged**